src/traits.jl

```diff
@@ -31,26 +31,83 @@ is the corresponding dimensions's axis. If `HasDimNames` is not defined for `x`
 default names are returned. `x` should have an `axes` method.
 
 ```jldoctest
-julia> using ImagesCore
+julia> using ImageCore
 
 julia> img = reshape(1:24, 2,3,4);
 
 julia> namedaxes(img)
-(dim_1 = Base.OneTo(2), dim_2 = Base.OneTo(3), dim_3 = Base.OneTo(4))
+(:row = Base.OneTo(2), :col = Base.OneTo(3), :page = Base.OneTo(4))
 ```
 """
-namedaxes(img::T) where T = namedaxes(HasDimNames(T), img)
+namedaxes(img::T) where T = _namedaxes(HasDimNames(T), img)
+_namedaxes(::HasDimNames{true}, img) = NamedTuple{names(img)}(axes(img))
+_namedaxes(::HasDimNames{false}, img) = NamedTuple{default_names(img)}(axes(img))
+
+default_names(img::AbstractArray{T,N}) where {T,N} = ntuple(i->default_name(i), N)
+
+@inline function default_name(i::Integer)
+    if i == 1
+        return :row
+    elseif i == 2
+        return :col
+    elseif i == 3
+        return :page
+    else
+        return Symbol(:dim_, i)
+    end
+end
 
-namedaxes(::HasDimNames{true}, x::T) where T = NamedTuple{names(x)}(axes(x))
+"""
+    findaxis(img, name) -> axis
 
-function namedaxes(::HasDimNames{false}, img::AbstractArray{T,N}) where {T,N}
-    NamedTuple{default_names(Val(N))}(axes(img))
+Returns the axis (as in `axes(img, dim)`) that corresponds to the `name`. If no
+matching name is found any empy axis (i.e. `0:0`) is returned.
```

Comment:
```suggestion
matching name is found any empty axis (i.e. `0:-1`) is returned.
```

I almost wonder if `axisrange` is a better name for this operation?

Reply:
In this case would you be thinking of the `axis` as the names and `range` as the tuple of `axes`?

Reply:
I'm just trying to ask myself, "what would I expect `findaxis` to do?" And I am not sure. `findmin` returns both the value and location of the minimum. So maybe `findaxis` should return the value (i.e., range) and dimension index (like `finddim`)? Or maybe `axisvalue`, which is again back to the names chosen by AxisArrays.

Reply:
I guess I'm trying to imitate the behavior of `axes(img, i)` where `i` would be a symbol. The trick is that whatever type `img` is may not wrap axes that are also directly linked to their names (e.g., one array type stores names wraps another type that stores axes). So if we can access the axes and names independently then we can use `namedaxes` to connect these otherwise independent features.

I think the rationale of `findaxis` returning something similar to `findmin` makes a lot of sense though. In this case maybe `findaxis` would return something like `NamedTuple{(:dim_1)}((1:10,))` and `axisvalue` would return the individual axis/range.

Reply:
I could go with that latter proposal.